### PR TITLE
Migrate emojiStore to use request.CTX instead of context.Context

### PR DESCRIPTION
**Pull request by fmartingr**
#### Summary
Migrate `EmojiStore` methods that make use of a `context.Context` to use `request.CTX` instead.

#### Ticket Link
..

#### Release Note
```release-note
Migrate EmojiStore to use request.CTX
```

### CODE REVIEW

**Comment by fmartingr** [comment]
Looking for feedback on the changes performed. Switching to the `request.CTX` object involves more changes that just using the new object because the methods where it's used 1) may not have `request.CTX` to being with, so we need to propagate this (not in this PR, thankfully) 2) use `context.Context` in other scenarios (see the `WithMaster` changes).

File: server/public/shared/request/context.go [hunk 1]
```diff
@@ -46,6 +46,13 @@ func EmptyContext(logger mlog.LoggerIFace) *Context {
 	}
 }
 
+// TestContext creates an empty context with a new logger to use in testing where a test helper is
+// not required.
+func TestContext() *Context {
+	logger, _ := mlog.NewLogger()
```
Comment (agnivade) [inline server/public/shared/request/context.go:52]: `mlog.CreateConsoleTestLogger` is a better choice I think

File: server/channels/store/sqlstore/context.go [hunk 2]
```diff
@@ -23,6 +25,12 @@ func WithMaster(ctx context.Context) context.Context {
 	return context.WithValue(ctx, storeContextKey(useMaster), true)
```
Comment (agnivade) [inline server/channels/store/sqlstore/context.go:25]: We should add a comment that this function is now deprecated and ask users to use the new one. And when all instances of `WithMaster` have been replaced, we should make it a private function.

File: server/public/shared/request/context.go (hunk 1 above)
Comment (fmartingr) [reply server/public/shared/request/context.go:52]: Isn't `mlog.CreateTestLogger` better? Or do we want logging to stdout everything on tests?

Comment (agnivade) [reply server/public/shared/request/context.go]: 0/5 from my side. Whatever you think is the bets.

**Review by agnivade — APPROVED** [review]

File: server/public/shared/request/context.go (hunk 1 above)
Comment (fmartingr) [reply server/public/shared/request/context.go]: `CreateTestLogger` is not being used, so I'll stick with `CreateConsoleTestLogger` as suggested.

**Pull request merged**